### PR TITLE
chore: Updated documentation for XCFramework support

### PR DESCRIPTION
**Pull request by royjit**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,6 +45,15 @@ There are three ways to integrate the AWS Mobile SDK for iOS into your own proje
 
 You should use ONE and only one of these ways to import the AWS Mobile SDK. Importing the SDK in multiple ways loads duplicate copies of the SDK into the project and causes compiler/linker errors.
 
+> Note: If you are using XCFramework using Carthage or Dynamic Frameworks, `AWSMobileClient` is named as `AWSMobileClientXCF`. Import `AWSMobileClient` as this:
```
Comment (palpatim) [inline README.md:48]:
```suggestion
> Note: If you are using XCFrameworks (i.e., either Carthage or Dynamic Frameworks), the module `AWSMobileClient` is named as `AWSMobileClientXCF` to work around a [Swift issue](https://bugs.swift.org/browse/SR-11704). To use `AWSMobileClient`, import it as:
```

File: README.md [hunk 2]
```diff
@@ -86,6 +95,22 @@ For a complete list of our pods, check out the .podspec files in the root direct
 
 ### Carthage
 
+Carthage support XCFrameworks for XCode 12 or above follow the steps below to build a platform-independent xcframeworks:
```
Comment (palpatim) [inline README.md:98]:
```suggestion
Carthage support XCFrameworks for Xcode 12 or above. Follow the steps below to consume the AWS SDK for iOS using XCFrameworks:
```

File: README.md [hunk 3]
```diff
@@ -86,6 +95,22 @@ For a complete list of our pods, check out the .podspec files in the root direct
...
+
+1. Install the Carthage directly from source, support for XCode 12+ is not fully released in [Carthage](https://github.com/Carthage/Carthage#installing-carthage).
```
Comment (palpatim) [inline README.md:100]:
```suggestion
1. Install Carthage 0.37.0 or greater. (Carthage does not expect to support [XCFramework support until 0.37.0](https://github.com/Carthage/Carthage/releases/tag/0.36.1).) If it is not yet available, [install Carthage](https://github.com/Carthage/Carthage#installing-carthage) directly from source.
```

File: README.md [hunk 4]
```diff
@@ -123,7 +148,19 @@ For a complete list of our pods, check out the .podspec files in the root direct
 
 ### Frameworks
 
-1. Download the [latest SDK](https://sdk-for-ios.amazonwebservices.com/latest/aws-ios-sdk.zip). Older SDK versions can be downloaded from `https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-#.#.#.zip`, where `#.#.#` represents the version number. So for version 2.10.2, the download link is [https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.10.2.zip](https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.10.2.zip).
+#### XCFramework setup
+
+Starting AWS SDK iOS version 2.22.1, frameworks are released as XCFramework. Following the steps below to instal XCFramework.
```
Comment (palpatim) [inline README.md:153]:
```suggestion
Starting AWS SDK iOS version 2.22.1, SDK binaries are released as XCFrameworks. Follow the steps below to install XCFramework.
```

File: README.md [hunk 5]
```diff
@@ -123,7 +148,19 @@ For a complete list of our pods, check out the .podspec files in the root direct
...
+
+1. Download the [latest SDK](https://sdk-for-ios.amazonwebservices.com/latest/aws-ios-sdk.zip). Older SDK versions can be downloaded from `https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-#.#.#.zip`, where `#.#.#` represents the version number. So for version 2.22.1, the download link is [https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.22.1.zip](https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.22.1.zip).
+> Note: If you are using version < 2.22.1 please refer to the section below where frameworks are added.
+
+4. On your application targets’ General settings tab, in the Embedded Binaries section, drag and drop each xcframework you want to use from the downloaded folder.
```
Comment (palpatim) [inline README.md:158]:
```suggestion
2. Uncompress the ZIP file
3. On your application targets’ General settings tab, in the Embedded Binaries section, drag and drop each xcframework you want to use from the downloaded folder.
```


Also, will individual SDK ZIPs come when we do SPM support or is that something we need to document now?

File: README.md [hunk 6]
```diff
@@ -123,7 +148,19 @@ For a complete list of our pods, check out the .podspec files in the root direct
 
 ### Frameworks
 
-1. Download the [latest SDK](https://sdk-for-ios.amazonwebservices.com/latest/aws-ios-sdk.zip). Older SDK versions can be downloaded from `https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-#.#.#.zip`, where `#.#.#` represents the version number. So for version 2.10.2, the download link is [https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.10.2.zip](https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.10.2.zip).
+#### XCFramework setup
+
+Starting AWS SDK iOS version 2.22.1, frameworks are released as XCFramework. Following the steps below to instal XCFramework.
+
+1. Download the [latest SDK](https://sdk-for-ios.amazonwebservices.com/latest/aws-ios-sdk.zip). Older SDK versions can be downloaded from `https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-#.#.#.zip`, where `#.#.#` represents the version number. So for version 2.22.1, the download link is [https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.22.1.zip](https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.22.1.zip).
+> Note: If you are using version < 2.22.1 please refer to the section below where frameworks are added.
```
Comment (palpatim) [inline README.md:156]:
```suggestion
> Note: If you are using version < 2.22.1 please refer to the "Legacy framework setup" section below.
```

File: README.md [hunk 7]
```diff
@@ -123,7 +148,19 @@ For a complete list of our pods, check out the .podspec files in the root direct
...
+
+4. On your application targets’ General settings tab, in the Embedded Binaries section, drag and drop each xcframework you want to use from the downloaded folder.
+
+#### Legacy framework setup
+
+1. Download the required SDK using `https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-#.#.#.zip`, where `#.#.#` represents the version number. So for version 2.10.2, the download link is [https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.10.2.zip](https://sdk-for-ios.amazonwebservices.com/aws-ios-sdk-2.10.2.zip).
+> Note: If you are using version > 2.22.0 please refer to the section above where xcframeworks are added.
```
Comment (palpatim) [inline README.md:163]:
```suggestion
> Note: If you are using version > 2.22.0 please refer to the "XCFramework setup" section above. 
```

File: README.md [hunk 8]
```diff
@@ -170,14 +207,7 @@ When we release a new version of the SDK, you can pick up the changes as describ
...
 
-1. In Xcode's **Project Navigator**, type "AWS" to find the AWS frameworks that were manually added to your project.  Manually select all of the AWS frameworks and hit **delete** on your keyboard. Then select **Move to Trash**.  If you were following the example from above which uses AWSMobileClient and AWSPinpoint, you would remove:
-
-    * `AWSAuthCore.framework`
-    * `AWSCognitoIdentityProvider.framework`
-    * `AWSCognitoIdentityProviderASF.framework`
-    * `AWSCore.framework`
-    * `AWSMobileClient.framework`
-    * `AWSPinpoint.framework`
+1. In Xcode's **Project Navigator**, type "AWS" to find the AWS frameworks that were manually added to your project.  Manually select all of the AWS frameworks and hit **delete** on your keyboard. Then select **Move to Trash**.
```
Comment (palpatim) [inline README.md:210]:
```suggestion
1. In Xcode's **Project Navigator**, type "AWS" to find the AWS Frameworks or XCFrameworks that you manually added to your project. Select all of the AWS Frameworks and hit **Delete** on your keyboard. Then select **Move to Trash**. 
```

File: README.md (hunk 5 above)
Comment (royjit) [reply README.md:158]: It is part of SPM, we can add it later.

File: README.md [hunk 9]
```diff
@@ -45,6 +45,15 @@ There are three ways to integrate the AWS Mobile SDK for iOS into your own proje
 
 You should use ONE and only one of these ways to import the AWS Mobile SDK. Importing the SDK in multiple ways loads duplicate copies of the SDK into the project and causes compiler/linker errors.
 
+> Note: If you are using XCFrameworks (i.e., either Carthage or Dynamic Frameworks), the module `AWSMobileClient` is named as `AWSMobileClientXCF` to work around a [Swift issue](https://bugs.swift.org/browse/SR-11704). To use `AWSMobileClient`, import it as:
+        
+        import AWSMobileClientXCF
+
+and use the it inside without the `XCF` suffix.
```
Comment (palpatim) [inline README.md:52]:
```suggestion
and use it your app code without the `XCF` suffix.
```

File: README.md [hunk 10]
```diff
@@ -86,6 +95,22 @@ For a complete list of our pods, check out the .podspec files in the root direct
 
 ### Carthage
 
+Carthage support XCFrameworks for Xcode 12 or above. Follow the steps below to consume the AWS SDK for iOS using XCFrameworks:
```
Comment (palpatim) [inline README.md:98]:
```suggestion

#### XCFrameworks (recommended)

Carthage supports XCFrameworks in Xcode 12 or above. Follow the steps below to consume the AWS SDK for iOS using XCFrameworks:
```

File: README.md [hunk 11]
```diff
@@ -86,6 +95,22 @@ For a complete list of our pods, check out the .podspec files in the root direct
...
+
+        github "aws-amplify/aws-sdk-ios"
+
+3. Then run the following command:
+    
+        $ carthage update --use-xcframeworks
+
+4. On your application targets’ General settings tab, in the Embedded Binaries section, drag and drop each xcframework you want to use from the Carthage/Build folder on disk.
+
+Building platform-specific framework bundles (Xcode 11 and below)
```
Comment (palpatim) [inline README.md:112]:
```suggestion
#### Frameworks with "fat libraries" (not recommended)

To build platform-specific framework bundles with multiple architectures in the binary, (Xcode 11 and below)
```

**Review by palpatim — APPROVED** [review]
Approved w/comments

**Pull request merged**